packages/db/index.js

```diff
@@ -218,6 +218,17 @@ export class DBClient {
    */
   async createUpload (data) {
     const now = new Date().toISOString()
+
+    const backupsObjects = data.backupUrls.map(url => ({
+      url,
+      created: data.created || now
+    }))
+
+    const backupsObjectsByURL = {}
+    backupsObjects.forEach(
+      backup => (backupsObjectsByURL[backup.url] = backup)
```

Comment:
nit: Can we just do the forEach on `data.backupUrls` and mutate `backupsObjectsByURL` as needed by creating the object here?

Reply:
why do we do this ? can't this be just a list of urls ?

Reply:
what if `backups_urls` is just an `text[]`, this function send only one url per call and the pg create upload function just appends to the array ?

Reply:
We should also guarantee URLs are unique

Reply:
yes, because it was not appending just overwriting

Reply:
I went with a JSONB structure to keep the creation timestamp and handle unique URLs. This also adds the possibility of adding extra metadata to the backup in the future.

We could call `createUpload` multiple times with the same backup URL. I'm unsure how to enforce unique URLs when appending to the array (maybe with unnest and aggregation).

I'm new to the project, and I'm unsure why `createUpload` accepts an array for `backupUrls`. From an existing test, it looks like it is to support creating backups in different locations/regions. It could be refactored to accept only one URL.

Reply:
We dont need any extra metadata about these urls in fact they will go away with uploads v2, we need them now just as a reference to be able to do migrations etc.

Create upload in the api and pg function should only accept one url, even when we chunk we just call create upload multiple times with the same root CID.

I get that jsonb merge is nice here, but we need to think about nft.storage and how we are going to fix it. In nft.storage we already have an array, the easiest migration path is to just append to it instead of migrating to a new structure.

Having only unique urls is nice but not critical because in any situation where we actually need to read those we can just select distinct from the array.

Reply:
@hugomrdias, ok, thank you, this makes sense now. I've made the changes to use text[] instead.